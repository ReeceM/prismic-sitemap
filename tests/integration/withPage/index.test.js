const {
  nextBuild,
  startApp,
  renderViaHTTP
} = require('../next-test-utils');

jasmine.DEFAULT_TIMEOUT_INTERVAL = 1000 * 80;

describe('Using Sitemap Generator', () => {
  it('loads sitemap.xml', async () => {
    await nextBuild(__dirname);

    await startApp(__dirname, ['-p', '44343'], { stdout: true, stderr: true })

    const html = await renderViaHTTP(44343, '/sitemap.xml');

    expect(html).toContain('xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"');

    expect(html).toMatchSnapshot();

  });
});
