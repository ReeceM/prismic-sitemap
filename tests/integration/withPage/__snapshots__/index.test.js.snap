// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using Sitemap Generator visiting page loads sitemap.xml 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:news=\\"http://www.google.com/schemas/sitemap-news/0.9\\" xmlns:xhtml=\\"http://www.w3.org/1999/xhtml\\" xmlns:image=\\"http://www.google.com/schemas/sitemap-image/1.1\\" xmlns:video=\\"http://www.google.com/schemas/sitemap-video/1.1\\"><url><loc>https://example.com/page/a-demo-page</loc><changefreq>monthly</changefreq><priority>1.0</priority></url><url><loc>https://example.com/blog/a-demo-blog-post</loc><changefreq>weekly</changefreq><priority>0.8</priority></url></urlset>"`;
